prow/cmd/checkconfig/main.go

```diff
@@ -1052,23 +1052,45 @@ func verifyOwnersPlugin(cfg *plugins.Configuration) error {
 	return nil
 }
 
+func checkRestrictedLabelNotInAdditionalLables(restricted string, label plugins.Label) bool {
```

Comment:
Let's have this as a `Label{}` method in https://github.com/kubernetes/test-infra/blob/d0d3c6d7a8d0/prow/plugins/config.go#L381

Also the name can be something like `isRestrictedLabelInAdditionalLables()` ?